xds/src/main/java/io/grpc/xds/XdsClientImpl.java

```diff
@@ -230,14 +233,15 @@ private void cleanUpResources() {
 
   @Override
   void watchConfigData(String hostName, int port, ConfigWatcher watcher) {
-    checkState(configWatcher == null, "ConfigWatcher is already registered");
+    checkState(configWatcher == null, "watcher for %s already registered", hostName);
```

Comment:
nit: Watcher

Reply:
Ah... I guess I've already messed up the format of Precondition error messages. Previously I always put whole sentences (i.e., starting with capitalized word and ending with a period). But latter I found that verbose and saw other places in our codebase put error messages as concise as possible. It doesn't need to be grammatically correct and should skip words like "a/an", "the", etc. So I decided to follow that style. 

For log messages, they still starts with capitalized word. But I may not put a period, as sometimes as period may be misleading (e.g., when the log message ends with and IP address).